runtime/gc_glue_java/MarkingDelegate.cpp

```diff
@@ -272,7 +272,8 @@ MM_MarkingDelegate::scanContinuationNativeSlots(MM_EnvironmentBase *env, omrobje
 		bStackFrameClassWalkNeeded = isDynamicClassUnloadingEnabled();
 #endif /* J9VM_GC_DYNAMIC_CLASS_UNLOADING */
 
-		GC_VMThreadStackSlotIterator::scanSlots(currentThread, objectPtr, (void *)&localData, stackSlotIteratorForMarkingDelegate, bStackFrameClassWalkNeeded, false);
+		bool bNeedMutex = _extensions->enableContinuationMountingMutex && _extensions->shouldScavengeNotifyGlobalGCOfOldToOldReference();
```

Comment:
unfortunately we will either have to introduce a new API that is not Scav specific (what is OMR change) or find another existing way to check this (via ConcurrentGC mode?)